File: Examples/VisualBasic/Text/RetrieveBulletOrNumberList.vb
Imports System.Globalization
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic
Namespace Text
    Public Class RetrieveBulletOrNumberList
        Public Shared Sub Run()
            ' ExStart:RetrieveBulletOrNumberList
            Dim dataDir As String = RunExamples.GetDataDir_Text()

            ' Load the document into Aspose.Note.
            Dim oneFile As New Document(dataDir & Convert.ToString("ApplyNumberingOnText.one"))

            ' Retrieve a collection nodes of the outline element
            Dim nodes As IList(Of OutlineElement) = oneFile.GetChildNodes(Of OutlineElement)()

            ' Iterate through each node
            For Each node As OutlineElement In nodes
                If node.NumberList IsNot Nothing Then
                    Dim list As NumberList = node.NumberList
                    ' Retrieve font name
                    Console.WriteLine("Font Name: " + list.Font)
                    ' Retrieve font length
                    Console.WriteLine("Font Length: " + list.Font.Length.ToString())
                    ' Retrieve font size
                    Console.WriteLine("Font Size: " + list.FontSize.ToString())
                    ' Retrieve font color
                    Console.WriteLine("Font Color: " + list.FontColor.ToString())
                    ' Retrieve format
                    Console.WriteLine("Font format: " + list.Format)
                    ' Check bold
                    Console.WriteLine("Is bold: " + list.IsBold.ToString())
                    ' Check italic
                    Console.WriteLine("Is italic: " + list.IsItalic.ToString())
                    Console.WriteLine()
                End If
            Next
            ' ExEnd:RetrieveBulletOrNumberList           
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Text/RetrieveBulletOrNumberList.cs
using System;
using System.Globalization;
using System.IO;
using Aspose.Note;
using System.Collections.Generic;

namespace Aspose.Note.Examples.CSharp.Text
{
    public class RetrieveBulletOrNumberList
    {
        public static void Run()
        {
            // ExStart:RetrieveBulletOrNumberList
            // ExFor:OutlineElement
            // ExFor:OutlineElement.NumberList
            // ExFor:NumberList
            // ExFor:NumberList.Font
            // ExFor:NumberList.FontSize
            // ExFor:NumberList.FontColor
            // ExFor:NumberList.Format
            // ExFor:NumberList.IsBold
            // ExFor:NumberList.IsItalic
            // ExSummary:Shows how to retrieve information about list's formatting.

            string dataDir = RunExamples.GetDataDir_Text();

            // Load the document into Aspose.Note.
            Document oneFile = new Document(dataDir + "ApplyNumberingOnText.one");

            // Retrieve a collection nodes of the outline element
            IList<OutlineElement> nodes = oneFile.GetChildNodes<OutlineElement>();

            // Iterate through each node
            foreach (OutlineElement node in nodes)
            {
                if (node.NumberList != null)
                {
                    NumberList list = node.NumberList;
                    
                    // Retrieve font name
                    Console.WriteLine("Font Name: " + list.Font);
                    
                    // Retrieve font length
                    Console.WriteLine("Font Length: " + list.Font.Length);
                    
                    // Retrieve font size
                    Console.WriteLine("Font Size: " + list.FontSize);
                    
                    // Retrieve font color
                    Console.WriteLine("Font Color: " + list.FontColor);
                    
                    // Retrieve format
                    Console.WriteLine("Font format: " + list.Format);
                    
                    // Check bold
                    Console.WriteLine("Is bold: " + list.IsBold);
                    
                    // Check italic
                    Console.WriteLine("Is italic: " + list.IsItalic);
                    Console.WriteLine();
                }
            }

            // ExEnd:RetrieveBulletOrNumberList           
        }
    }
}
